blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md

```diff
@@ -0,0 +1,797 @@
+    <div className="App">
+      <Home cUSDBalance={cUSDBalance} addFreelancer={addFreelancer} />
+      <Freelancers
+        freelancers={freelancers}
+        hireFreelancer={hireFreelancer}
+        changeDescription={changeDescription}
+        changePrice={changePrice}
+        walletAddress={address}
+        toggleAvailable={toggleAvailable}
+      />
+    </div>
+  );
+}
+
+export default App;
+```
+
+### Breakdown
+
+The first few lines of the code import the required dependencies and components.
+
+```javascript
+import "./App.css";
+import Home from "./components/home";
+import { Freelancers } from "./components/Freelancers";
+import { useState, useEffect, useCallback } from "react";
+import Web3 from "web3";
+import { newKitFromWeb3 } from "@celo/contractkit";
+import freelancer from "./contracts/freelancer.abi.json";
+import IERC from "./contracts/IERC.abi.json";
+import BigNumber from "bignumber.js";
+
+```
+
+#### Declaring constants and initializing state:
+
+We declare some constants for our contracts' addresses and decimals. We also initialize our state using useState hooks.
+
+```javascript
+const ERC20_DECIMALS = 18;
+const contractAddress = "0xA371b0718b07026D87F2606295562e86Fa3Dd364";
+const cUSDContractAddress = "0x874069Fa1Eb16D44d622F2e0Ca25eeA172369bC1";
+
+function App() {
+  const [contract, setcontract] = useState(null);
+  const [address, setAddress] = useState(null);
+  const [kit, setKit] = useState(null);
+  const [cUSDBalance, setcUSDBalance] = useState(0);
+  const [freelancers, setFreelancers] = useState([]);
+  // ...
+}
+```
+
+#### Connecting to the user's wallet and get user's balance:
+
+```javascript
+const connectToWallet = async () => {
+    if (window.celo) {
+      try {
+        await window.celo.enable();
+        const web3 = new Web3(window.celo);
+        let kit = newKitFromWeb3(web3);
+
+        const accounts = await kit.web3.eth.getAccounts();
+        const user_address = accounts[0];
+        kit.defaultAccount = user_address;
+
+        await setAddress(user_address);
+        await setKit(kit);
+      } catch (error) {
+        console.log(error);
+      }
+    } else {
+      alert("Error Occurred");
+    }
+  };
+
+  const getBalance = useCallback(async () => {
+    try {
+      const balance = await kit.getTotalBalance(address);
+      const USDBalance = balance.cUSD.shiftedBy(-ERC20_DECIMALS).toFixed(2);
+
+      const contract = new kit.web3.eth.Contract(freelancer, contractAddress);
+      setcontract(contract);
+      setcUSDBalance(USDBalance);
+    } catch (error) {
+      console.log(error);
+    }
+  }, [address, kit]);
+```
+
+We create a `connectToWallet` function that checks if the Celo extension is available, then connects to the user's wallet, and sets the user's default account.
+
+We also create a `getBalance` function using useCallback to fetch the user's cUSD balance and initialize the freelancer contract.
+
+#### Funtions to interact with the smart contract:
```

Comment:
can you make this a double hashtag or triple at most or maybe just use a bold text